plugins/out_s3/s3.c

```diff
@@ -431,6 +482,25 @@ static int cb_s3_init(struct flb_output_instance *ins,
     if (tmp) {
         ctx->canned_acl = (char *) tmp;
     }
+
+    tmp = flb_output_get_property("compression", ins);
+    if (tmp) {
+        if (strcmp((char *) tmp, "gzip") != 0) {
+            flb_plg_error(ctx->ins, 
+                          "'gzip' is currently the only supported value for 'compression'");
+            return -1;
+        } else if (ctx->use_put_object == FLB_FALSE) {
+            flb_plg_error(ctx->ins, "use_put_object should be enabled");
```

Comment:
This does not give enough context: how about: "use_put_object must be enabled when compression is enabled"

Reply:
Sure.